azerothcore/azerothcore-wotlk
# Fix(Script/scholo): Improvements for Kirtonos

## Changes Proposed:
- Removed most SAI from brazier, it now only sets the instance data
- Event starts at the end of the gameobject use, not at the start. ie he spawns when the brazier is lit
- Gate and brazier control is moved from kirtonos.cpp to scholomance.cpp
- Retry is now possible after a wipe, granted one has enough blood.
- Kirtonos doesn't swim in the air in humanoid form anymore

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8397
- closes https://github.com/azerothcore/azerothcore-wotlk/issues/8386
- closes https://github.com/azerothcore/azerothcore-wotlk/issues/8852 
- closes https://github.com/chromiecraft/chromiecraft/issues/2221
- closes https://github.com/chromiecraft/chromiecraft/issues/2041

## SOURCE:


## Tests Performed:

- tested on my own machine. https://drive.google.com/file/d/1h2Sc_UCuLoGhe9s-WOogzoRG_WeluJnq/view?usp=sharing
- tested fail event (enabling GM mode), door opens and brazier reset
- tested 2 players clicking at the same time, only one spawns and the last clicker gets "invalid target" upon completion, which afaik is blizzlike
- tested killing the boss, door opens brazier doesn't reset


## How to Test the Changes:


1. Go to scholo at kirtonos spawn, use blood of innocent (.add 13523)
2. Test the event in usual ways: wipe, boss kill, etc (note that if the boss is killed and you respawn him, the event probably wont work anymore)
3. Test what happens if you get thrown out the window too, should just cause a reset but good to be sure.

## Known Issues and TODO List:


- [ ] Might have broken the "load after crash" feature.
- [ ] I broke the flying path slightly towards the end where he lands I believe, but it's still good enough.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
